npm/private/list_sources.bzl

```diff
@@ -0,0 +1,19 @@
+"Output a list of source files to a file"
+
+def _impl(ctx):
+    output = ctx.actions.declare_file(ctx.label.name + ".txt")
+    content = "\n".join([file.short_path for file in ctx.files.srcs]) + "\n"
```

Comment:
lets use file.path here so its obvious in the output if a generated files sneaks in here. for source files in the user workspace short_path and path are equivalent.